module/VuFind/src/VuFind/Auth/OpenIDConnect.php

```diff
@@ -552,7 +552,10 @@ protected function getJwks(): array
             }
             $jwks = json_decode($response->getBody(), true);
             foreach ($jwks['keys'] as $i => $jwk) {
-                $this->jwks[$jwk['kid'] ?? $i] = $jwk;
+                // Filter for signature keys only.
+                if (isset($jwk['use']) && $jwk['use'] === 'sig') {
```

Comment:
A slightly less verbose variation:

```suggestion
                if (($jwk['use'] ?? '') === 'sig') {
```